great_expectations/experimental/datasources/interfaces.py

```diff
@@ -70,13 +90,203 @@ def datasource(self, ds: Datasource):
         self._datasource = ds
 
-    def get_batch_request(self, options: Optional[BatchRequestOptions]) -> BatchRequest:
+    def batch_request_options_template(
+        self,
+    ) -> BatchRequestOptions:
+        """A BatchRequestOptions template for get_batch_request.
+
+        Returns:
+            A BatchRequestOptions dictionary with the correct shape that get_batch_request
+            will understand. All the option values are defaulted to None.
+        """
+        template: BatchRequestOptions = {}
+        if not self.column_splitter:
+            return template
+        return {p: None for p in self.column_splitter.param_names}
+
+    def get_batch_request(
+        self, options: Optional[BatchRequestOptions] = None
+    ) -> BatchRequest:
+        """A batch request that can be used to obtain batches for this DataAsset.
+
+        Args:
+            options: A dict that can be used to limit the number of batches returned from the asset.
+                The dict structure depends on the asset type. A template of the dict can be obtained by
+                calling batch_request_options_template.
+
+        Returns:
+            A BatchRequest object that can be used to obtain a batch list from a Datasource by calling the
+            get_batch_list_from_batch_request method.
+        """
+        if options is not None and not self._valid_batch_request_options(options):
+            raise BatchRequestError(
+                "Batch request options should have a subset of keys:\n"
+                f"{list(self.batch_request_options_template().keys())}\n"
+                f"but actually has the form:\n{pf(options)}\n"
+            )
+        return BatchRequest(
+            datasource_name=self.datasource.name,
+            data_asset_name=self.name,
+            options=options or {},
+        )
+
+    def _valid_batch_request_options(self, options: BatchRequestOptions) -> bool:
+        return set(options.keys()).issubset(
+            set(self.batch_request_options_template().keys())
+        )
+
+    def get_batch_list_from_batch_request(
+        self, batch_request: BatchRequest
+    ) -> List[Batch]:
+        """A list of batches that match the BatchRequest.
+
+        Args:
+            batch_request: A batch request for this asset. Usually obtained by calling
+                get_batch_request on the asset.
+
+        Returns:
+            A list of batches that match the options specified in the batch request.
+        """
+        # We translate the batch_request into a BatchSpec to hook into GX core.
+        self._validate_batch_request(batch_request)
+
+        batch_list: List[Batch] = []
+        for request in self._fully_specified_batch_requests(batch_request):
+            batch_list.append(self._batch_from_fully_specified_batch_request(request))
+        self._sort_batches(batch_list)
+        return batch_list
+
+    def _batch_from_fully_specified_batch_request(self, request: BatchRequest):
```

Comment:
This is a new abstract method which encapsulates how a particular `DataAsset` creates a `Batch`. It's possible this could be generalized further and we may once we start implementing the pandas datasource.